js/BUILDING.md

```diff
@@ -1,184 +1,121 @@
-# Ice for JavaScript Build Instructions
+# Building Ice for JavaScript from source
 
-This file describes how to build and install Ice for JavaScript from source code. If you prefer, you can also download
-[binary distributions] for the supported platforms.
+## Table of contents
 
-<!-- TOC depthFrom:2 -->
+- [Building Ice for JavaScript from source](#building-ice-for-javascript-from-source)
+  - [Table of contents](#table-of-contents)
+  - [Prerequisites](#prerequisites)
+  - [Build roadmap](#build-roadmap)
+  - [Building Ice for JavaScript](#building-ice-for-javascript)
+  - [Running the tests](#running-the-tests)
+  - [Creating the NPM package](#creating-the-npm-package)
+    - [Slice compilers](#slice-compilers)
+  - [Generating the API reference](#generating-the-api-reference)
 
-- [JavaScript Build Requirements](#javascript-build-requirements)
-- [Building the JavaScript libraries and NodeJS packages](#building-the-javascript-libraries-and-nodejs-packages)
-- [Running the JavaScript Tests](#running-the-javascript-tests)
-  - [Browser Information](#browser-information)
-    - [Self-Signed Certificate](#self-signed-certificate)
-    - [Secure WebSockets on iOS and Android](#secure-websockets-on-ios-and-android)
-      - [Installing Certificates on iOS](#installing-certificates-on-ios)
-      - [Installing Certificates on Android](#installing-certificates-on-android)
-- [Installing a Source Build](#installing-a-source-build)
+## Prerequisites
 
-<!-- /TOC -->
+1. Node.js 22 or later.
 
-## JavaScript Build Requirements
+2. The Slice to JavaScript compiler (slice2js) from the C++ source distribution. \
+   Refer to the [build instructions](../cpp/BUILDING.md) in the `cpp` folder for details on how to build the C++ source
+   distribution.
 
-To build Ice for JavaScript you must have the following:
+3. Python 3.12 is required for running the tests. The Glacier2 test also require the `passlib` Python package.
 
-- The `slice2js` compiler from Ice for C++. If you have not built Ice for C++ in this source distribution, refer to the
-  [C++ build instructions](../cpp/BUILDING.md).
-- Node.js 22 or later
+4. Ice for Python is required for running the JavaScript in a web browser.\
+   Refer to the [build instructions](../python/BUILDING.md) in the `python` folder for details on how to build the
```

Comment:
Probably not a good advice as you'd typically use the Python wheel (possibly nightly wheel).

Reply:
Should we mention how to install it, or is enough to say it is required?

Reply:
I think if a user is doing a source build they should just do a source build for everything.